data/lua/as_text.lua

```diff
@@ -0,0 +1,66 @@
+-- NOTE: the string output from here is intended solely as an aid in debugging; it should never be taken and used as an input to anything else.
+
+local delete_chars = string.char(00, 01, 02, 03, 04, 05, 06, 07, 08, 09, 10, 11, 12, 13, 14, 15,
+	16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31) -- https://www.ietf.org/rfc/rfc4627.txt
+local delete_regexp = "[" .. delete_chars .. "]"
+
+-- escaping takes 3/4 of the time, but we can't avoid it...
+local function escape(str)
+	str = string.gsub(str, delete_regexp, "")
```

Comment:
Updated to `"%c"`.  Performance shouldn't be an issue here I don't think since this is only meant to be used for debugging output, and it's not exceptionally slow as-is.